api/http/handler/hostmanagement/fdo/configure.go

```diff
@@ -0,0 +1,181 @@
+package fdo
+
+import (
+	"encoding/hex"
+	"errors"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+
+	cbor "github.com/fxamacker/cbor/v2"
+
+	httperror "github.com/portainer/libhttp/error"
+	"github.com/portainer/libhttp/request"
+	"github.com/portainer/libhttp/response"
+	"github.com/sirupsen/logrus"
+)
+
+type deviceConfigurePayload struct {
+	EdgeKey string `json:"edgekey"`
+	Name    string `json:"name"`
+	Profile int    `json:"profile"`
+}
+
+func (payload *deviceConfigurePayload) Validate(r *http.Request) error {
+	if payload.EdgeKey == "" {
+		return errors.New("invalid edge key provided")
+	}
+
+	if payload.Name == "" {
+		return errors.New("the device name cannot be empty")
+	}
+
+	return nil
+}
+
+// @id fdoConfigureDevice
+// @summary configure an FDO device
+// @description configure an FDO device
+// @description **Access policy**: administrator
+// @tags intel
+// @security jwt
+// @produce json
+// @param edgekey query string true "Environment edge key"
+// @param name query string true "Environment name"
+// @param profile query string true "FDO dployment profile name"
```

Comment:
```suggestion
// @param profile query string true "FDO deployment profile name"
```

Reply:
Taking a better look, I'm not sure if lines 44-46 are correct, those are not query params, I think it should be same as the `fdoConfigure` body param
```suggestion
// @param body body deviceConfigurePayload true "Device Configuration"
```
or something like that :)